File: ch1/prime100.py
# 20. 소수인지 판단하는 함수
def is_prime(n):
    for i in range(2, n):
        print("{}, {}".format(n, i))
        if n % i == 0:
            return False
    return True
    
# count만큼 소수를 생성
def get_primes(count):
    res = []
    i = 2
    while len(res) < count :
        if is_prime(i):
            res.append(i)
        i += 1
    return res

print(get_primes(10))

File: ch1/prime100.rs
// 20. 소수인지 판단하는 함수
fn is_prime(n: usize) -> bool{
    for i in 2..n {
        if n % i == 0 {
            return false;
        }
    }
    return true;
}

fn get_prime(primes: &mut[usize; 100]){
    let mut i = 2;
    let mut count = 0;

    while count < 100 {
        if is_prime(i) {
            primes[count] = i;
            count += 1;
        }
        i += 1;
    }
}

fn main(){
    let mut primes = [0; 100];
    get_prime(&mut primes);
    println!("{:?}", primes);
}